__tests__/feature/previewRdf.test.js

```diff
@@ -0,0 +1,47 @@
+// Copyright 2020 Stanford University see LICENSE for license
+
+import { renderApp } from 'testUtils'
+import { fireEvent, screen } from '@testing-library/react'
+import * as sinopiaServer from 'sinopiaServer'
+import { getFixtureResourceTemplate } from 'fixtureLoaderHelper'
+import Config from 'Config'
+
+jest.mock('sinopiaServer')
+global.$ = jest.fn().mockReturnValue({ popover: jest.fn() })
+jest.spyOn(Config, 'useResourceTemplateFixtures', 'get').mockReturnValue(true)
+
+const rdf = `<> <http://id.loc.gov/ontologies/bibframe/uber/template1/property7> "Default literal1", "Default literal2";
+    <http://id.loc.gov/ontologies/bibframe/uber/template1/property8> <http://sinopia.io/defaultURI1>, <http://sinopia.io/defaultURI2>;
+    <http://sinopia.io/vocabulary/hasResourceTemplate> "resourceTemplate:testing:uber1";
+    a <http://id.loc.gov/ontologies/bibframe/Uber1>.
+<http://sinopia.io/defaultURI1> <http://www.w3.org/2000/01/rdf-schema#label> "Default URI1".`
+
+
+describe('preview RDF after editing', () => {
+  sinopiaServer.foundResourceTemplate.mockResolvedValue(true)
+  sinopiaServer.getResourceTemplate.mockImplementation(getFixtureResourceTemplate)
+
+  it('adds properties and then displays preview RDF model', async () => {
+    renderApp()
+
+    // Open the editor and then the templates tab
+    fireEvent.click(screen.getByRole('link', { name: 'Linked Data Editor' }))
+    fireEvent.click(screen.getByRole('link', { name: 'Resource Templates' }))
+
+    // Click an existing resource template
+    await screen.findByText(/Uber template1/)
+    fireEvent.click(screen.getByRole('link', { name: 'Uber template1' }))
+
+    // Click on the Preview RDF Button
+    await screen.findByText(/Uber template1/)
+    fireEvent.click(screen.getAllByRole('button', { name: 'Preview RDF' })[0])
+
+    // Wait for RDF Preview Modal and selects turtle Format
+    await screen.findByText(/RDF Preview/)
+    fireEvent.change(screen.getByRole('combobox', { name: 'RDF Format Selection' }),
+      { target: { value: 'turtle' } })
+
+    // Tests for presence of turtle RDF in the model
+    expect(screen.findByText(rdf)).toBeTruthy()
```

Comment:
This returns a promise, so you either need an `await` or change to `getByText`. Also, the `expect` is not necessary, because it will throw if no match. See https://testing-library.com/docs/react-testing-library/cheatsheet#queries

Reply:
Good catch. I just updated the PR to use `getByTestId` to extract the `<pre>` and then compares the text content with the expected RDF string. Tried using `getByText` approach but was running into problems with selecting on the escaped HTML.